src/codegate/pipeline/secrets/signatures.py

```diff
@@ -215,16 +215,26 @@ def _load_signatures(cls) -> None:
             raise
 
     @classmethod
-    def find_in_string(cls, text: str) -> List[Match]:
-        """Search for secrets in the provided string."""
+    def find_in_string(cls, text: Union[str, List[str]]) -> List[Match]:
+        """Search for secrets in the provided string or list of strings."""
         if not text:
             return []
 
         if not cls._yaml_path:
             raise RuntimeError("SecretFinder not initialized.")
 
+        # Convert list to string if necessary (needed for Cline, which sends a list of strings)
+        if isinstance(text, list):
+            text = "\n".join(str(line) for line in text)
```

Comment:
This is converted to string to be later split to an array? Seems redundant. Maybe you want to pass it as-is if it's already an array?